lib/zendesk_apps_tools/server.rb

```diff
@@ -44,6 +43,18 @@ class Server < Sinatra::Base
       ZendeskAppsSupport::Installed.new([app_js], [installation]).compile_js
     end
 
+    get "/:file" do |file|
+      access_control_allow_origin
+      send_file File.join(settings.root, 'assets', file)
+    end
+
+    # This is for any preflight request
+    options "*" do
+      access_control_allow_origin
+      origin = request.env['HTTP_ORIGIN']
+      headers 'Access-Control-Allow-Headers' => request.env['HTTP_ACCESS_CONTROL_REQUEST_HEADERS'] if origin =~ ZENDESK_DOMAINS_REGEX
```

Comment:
And to make it more confusing it copies the value from yes another header called: Access-Control-Requests-Headers.